x-pack/plugins/reporting/server/core.ts

```diff
@@ -180,14 +184,58 @@ export class ReportingCore {
     return this.getPluginSetupDeps().elasticsearch;
   }
 
-  public async getSavedObjectsClient(fakeRequest: KibanaRequest) {
+  private async getSavedObjectsClient(request: KibanaRequest) {
     const { savedObjects } = await this.getPluginStartDeps();
-    return savedObjects.getScopedClient(fakeRequest) as SavedObjectsClientContract;
+    return savedObjects.getScopedClient(request) as SavedObjectsClientContract;
   }
 
   public async getUiSettingsServiceFactory(savedObjectsClient: SavedObjectsClientContract) {
     const { uiSettings: uiSettingsService } = await this.getPluginStartDeps();
     const scopedUiSettingsService = uiSettingsService.asScopedToClient(savedObjectsClient);
     return scopedUiSettingsService;
   }
+
+  public getSpaceId(request: KibanaRequest): string | undefined {
+    const spacesService = this.getPluginSetupDeps().spaces?.spacesService;
+    if (spacesService) {
+      const spaceId = spacesService?.getSpaceId(request);
+
+      if (spaceId !== DEFAULT_SPACE_ID) {
+        this.logger.info(`Request uses Space ID: ` + spaceId);
+        return spaceId;
+      } else {
+        this.logger.info(`Request uses default Space`);
+      }
+    }
+  }
+
+  public getFakeRequest(baseRequest: object, spaceId?: string) {
+    const fakeRequest = KibanaRequest.from({
+      path: '/',
+      route: { settings: {} },
+      url: { href: '/' },
+      raw: { req: { url: '/' } },
+      ...baseRequest,
+    } as Hapi.Request);
+
+    const spacesService = this.getPluginSetupDeps().spaces?.spacesService;
+    if (spacesService) {
+      if (spaceId && spaceId !== DEFAULT_SPACE_ID) {
+        this.logger.info(`Generating request for space: ` + spaceId);
+        this.getPluginSetupDeps().basePath.set(fakeRequest, `/s/${spaceId}`);
```

Comment:
cc @jportner Thanks for your help in figuring this part out! Please take a look at how I am using your advice and let me know if you have any thoughts about it.

Reply:
As discussed offline -- looks great!